src/app/features/data-import/data-import.service.ts

```diff
@@ -55,17 +57,45 @@ export class DataImportService {
     });
   }
 
-  async importCsvContentToDB(csv: string): Promise<void> {
+  async importCsvContentToDB(csv: string, importMeta: ImportMetaData): Promise<void> {
     const parsedCsv = this.parseCsvFile(csv);
 
+    // e.g. Child:abcd1234
+    const recordIdPrefix = importMeta.entityType + ":" + importMeta.transactionId;
+
+    // remove existing records, if any
+    // there is a chance of collision
+    const existingRecord = await this.db.getAll(recordIdPrefix);
+
+    for (const record of existingRecord) {
+      await this.db.remove(record);
+    }
+
+    let hasIdProperty = true;
+
+    if (!parsedCsv.meta.fields.includes("_id")) {
+      hasIdProperty = false;
+      parsedCsv.meta.fields.push("_id");
```

Comment:
In case the `_id` property is specified, shouldn't the `db.remove(record)` step be skipped? Because in this case the `transactionId` is irrelevant?

Reply:
Good question, if the transaction ID is expected to be unique, the search shouldn't find anything anyway - so calling the getAll shouldn't return anything, thus not deleting anything.

To change this, the workflow needs to be changed. The file needs to be parsed before it's passed to the import so one knows if there is an _id or not, if there is one, no transaction ID needs to be generated (which is currently always generated). So simply checking if there is an _id in the file and then not deleting something isn't correct either I think. What about having a file with _id and giving the transaction ID?

So changing the flow here is required to fully improve this.